Automattic/wp-calypso
# Marketplace search: Add highlighting to search results

#### Proposed Changes

* Use the `TextHighlight` component from  `@wordpress/components` to add highlighting functionality to search results
* Use the search term added by the user to highlight the components.
* The highlighting will be applied to:
  * Title
  * Author 
  * Description

#### Testing Instructions



- Go to `/plugins` and search for a plugin
- Check the search term is highlighted in the search results
- Go to `/plugins/<site-id>` and search for a plugin
- Check the search term is highlighted in the search results

Examples:

|Before | After Example 1| After Example 2|
|-------|------|------|
|        ![CleanShot 2022-08-04 at 12 11 45@2x](https://user-images.githubusercontent.com/3519124/182822617-a22164fb-71ce-4974-bf66-d602c34dc566.png)        |        ![CleanShot 2022-08-04 at 12 12 15@2x](https://user-images.githubusercontent.com/3519124/182822690-6cbc60dd-2e8c-4c11-bfe8-093d1a9a351f.png)        |        ![CleanShot 2022-08-04 at 12 14 04@2x](https://user-images.githubusercontent.com/3519124/182823090-2b60a4ff-88f3-42ca-8d70-b1f6bbbcf9fe.png)       |

#### Pre-merge Checklist






Related to #64574
